edx_repo_tools/oep2/explode_repos_yaml.py

```diff
@@ -48,62 +57,181 @@ def explode(hub, dry):
+                    )
+
+        pr_body = textwrap.dedent("""
+            This adds an `openedx.yaml` file, as described by OEP-2:
+            http://open-edx-proposals.readthedocs.io/en/latest/oeps/oep-0002.html
+
+            The data in this file was transformed from the contents of
+            edx/repo-tools-data:repos.yaml
+        """)
+        pr_title = 'Add an OEP-2 compliant openedx.yaml file'
+
+        existing_pr = [
+            pr
+            for pr
+            in gh_repo.iter_pulls(
+                head='edx:{}'.format(BRANCH_NAME),
+                state='open'
+            )
+        ]
+
+        if existing_pr:
+            pull = existing_pr[0]
+            if pull.title != pr_title or pull.body != pr_body:
+                dry_echo(
+                    dry,
+                    textwrap.dedent("""\
+                        Updated pull request {repo}#{number}: {title}
+                            URL: {url}
+                    """).format(
+                        url=pull.html_url,
+                        repo=gh_repo.full_name,
+                        number=pull.number,
+                        title=pull.title,
+                    ),
+                    fg='green'
+                )
+
+                if not dry:
+                    pull.update(
+                        title=pr_title,
+                        body=pr_body,
+                    )
+        else:
+            dry_echo(
+                dry,
+                textwrap.dedent("""\
+                    Created pull request {repo}#{number}: {title}
+                        URL: {url}
+                """).format(
+                    url=pull.html_url if not dry else "N/A",
+                    repo=gh_repo.full_name,
+                    number=pull.number if not dry else "XXXX",
+                    title=pull.title if not dry else pr_title,
+                ),
+                fg='green'
+            )
 
-        gh_repo.create_ref(
-            'refs/heads/{}'.format(BRANCH_NAME),
-            gh_repo.branch('master').commit.sha,
-        )
-        gh_repo.create_file(
-            path=OPEN_EDX_YAML,
-            message='Add an OEP-2 compliant openedx.yaml file',
-            content=file_contents,
-            branch=BRANCH_NAME,
-        )
-        pull = gh_repo.create_pull(
-            title='Add an OEP-2 compliant openedx.yaml file',
-            base='master',
-            head=BRANCH_NAME,
-        )
-        click.secho('Created pull request {} against {}'.format(
-            pull.html_url,
-            repo,
-        ), fg='green')
+            if not dry:
+                pull = gh_repo.create_pull(
+                    title=pr_title,
+                    base=gh_repo.default_branch,
+                    head=BRANCH_NAME,
+                    body=pr_body
+                )
 
 
 @click.command()
 @pass_github
 @click.option('--org', multiple=True, default=['edx', 'edx-ops'])
```

Comment:
And BTW, I can only see one repo in edx-ops, so I wonder if it should be in the default also.